### PR TITLE
Implements user search on Index

### PR DESCRIPTION
**Pull request by mattfeg**
Index can now return users that matches with param `name`.
Send a name for search as body params on index.
Example:
`{"name":"ABC"}`

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,7 +1,19 @@
 class UsersController < ApplicationController
   def index
     @users = User.all
-    render json: @users
+    searched_user = params[:name]
```
Comment (edimossilva) [inline app/controllers/users_controller.rb:4]: search_user_params is a better name

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
+
+    if searched_user.present?
+      filtered_users = @users.select { |user| user.name.downcase.include?(searched_user.downcase) }
```
Comment (edimossilva) [inline app/controllers/users_controller.rb:7]: extract this filter to a scope and make it at db level

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,7 +1,19 @@
...
+
+      if filtered_users.any?
+        render json: filtered_users
+      else
+        render json: { message: "Not found user" }, status: :not_found
```
Comment (edimossilva) [inline app/controllers/users_controller.rb:12]: we dont need not_found here, not found should be only at the show method, probably I confused myself when requested this to you or you missunderstood.

**Review by edimossilva — CHANGES_REQUESTED** [review]
- Fixes from my commments
- Add specs

**Review by mattfeg — COMMENTED** [review]
Fixes and tests requested done.

**Review by edimossilva — APPROVED** [review]

**Pull request merged**